docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,236 @@
+
+[source,sh]
+----
+kubectl logs -f elastic-operator-0 -n elastic-system
+----
+
+The operator is constantly attempting to reconcile Kubernetes resources to match the desired state.
+Logs with `INFO` level provide some insights about what is going on.
+
+There should be normally very few logs with `ERROR` level: they indicate something is not going as expected.
+
+Due to link:https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#concurrency-control-and-consistency[optimistic locking],
+there might be some errors reporting a conflict while updating a resource. Those are safe to ignore, since the update should go through at the next reconciliation attempt, which should happen almost immediately.
+
+[float]
+[id="{p}-eck-debug-logs"]
+=== Enable ECK debug logs
+
+To enable `DEBUG` level logs on the operator, restart it with the flag `--enable-debug-logs=true`. For example:
+
+[source,sh]
+----
+kubectl edit statefulset.apps -n elastic-system elastic-operator
+----
+
+and change the following lines from:
+
+[source,yaml]
+----
+  spec:
+    containers:
+    - args:
+      - manager
+      - --operator-roles
+      - all
+      - --enable-debug-logs
+      - false
+----
+
+to
+
+[source,yaml]
+----
+  spec:
+    containers:
+    - args:
+      - manager
+      - --operator-roles
+      - all
+      - --enable-debug-logs
+      - true
+----
+
+[float]
+[id="{p}-pause-controllers"]
+=== Pause ECK controllers
+
+When debugging Elasticsearch, it may be necessary to "pause" the operator reconciliations so no resource gets modified or created in the meantime.
+
+It can be achieved by setting the annotation `common.k8s.elastic.co/pause` to `true` to any resource controlled by the operator :
+
+- Elasticsearch
+- Kibana
+- ApmServer
+
+[source,yaml]
+----
+metadata:
+  annotations:
+    common.k8s.elastic.co/pause: "true"
+----
+
+Or in one line:
+
+[source,sh]
+----
+kubectl annotate elasticsearch quickstart --overwrite common.k8s.elastic.co/pause=true
+----
+
+[float]
+[id="{p}-get-k8s-events"]
+=== Get Kubernetes events
+
+ECK will emit some events when:
+
+* important operations are performed (example: a new Elasticsearch Pod was created)
+* something is wrong, and the user should know about it
+
+Fetch Kubernetes events:
+
+[source,sh]
+----
+kubectl get events
+
+(...)
+28s       25m       58        elasticsearch-sample-es-p45nrjch29.15b3ae4cc4f7c00d   Pod                             Warning   FailedScheduling    default-scheduler                                         0/3 nodes are available: 1 node(s) had no available volume zone, 2 Insufficient memory.
+28s       25m       52        elasticsearch-sample-es-wxpnzfhqbt.15b3ae4d86bc269f   Pod                             Warning   FailedScheduling    default-scheduler                                         0/3 nodes are available: 1 node(s) had no available volume zone, 2 Insufficient memory.
+----
+
+The list of events can be filtered to a particular Elasticsearch cluster:
```

Comment:
`can be filtered to show only those relevant for a particular Elasticsearch cluster`